README.md

```diff
@@ -1,9 +1,536 @@
+
+* `onCreate` is where you will instantiate your variables, and where you can call some methods to access IOs.
+* `onDestroy` is where you can free / close your IO resources
+
+Methods that deal with IO may throw an `IOException`. You will have to catch this checked exception, both in the onCreate and onDestroy:
+
+```java
+try {
+  // access IOs here
+} catch (IOException e) {
+  Log.e(TAG, "IO issue", e);
+}
+```
+
+### How to get started with Android Things from scratch?
+
+To save you some time, we have already prepared a sample project, flashed Android Things on Raspberry Pi boards, and set up the Wi-Fi.  
+If you want to repeat this at home, here is what we did:
+
+- [Download and flash Android Things on the Raspberry Pi 3](https://developer.android.com/things/hardware/raspberrypi.html)
+- [Clone the Android Things New Project Template](https://github.com/androidthings/new-project-template)
+- Add dependencies in the `build.gradle` file
+
+That's all!  
+We've also built a paper catapult for you. How-to video [here](https://www.youtube.com/watch?v=9JvV8PWawLs)
+
+
+## Hardware
+
+### Discovering the Raspberry Pi
+
+The following pinout diagram illustrates the locations of the available ports exposed by the breakout connectors of the Raspberry Pi:
+
+![Raspberry Pi pinout][raspberrypi-pinout]
+
+What this means basically is that you can connect up to 2 PWM devices, up to 15 GPIO devices, the Raspberry Pi also has 1 port to connect I²C, SPI and UART devices.  
+Each of these interfaces has a name (e.g.: "BCM4", "UART0", "PWM0"). When you connect a component (e.g. an LED) to the board, you will need to know the name of the connector to communicate with the component.
+
+### Discovering the RainbowHAT
+
+The RainbowHAT is a board that features sensors, inputs, and displays you can directly plug onto the Raspberry Pi.  
+It makes it easy to get started with Android Things, skipping the (fun, but time consuming) components wiring part.
+
+Here is a list of the different RainbowHAT components, and their associated connector names:
+
+* **Red LED**: "BCM6"
+* **Green LED**: "BCM19"
+* **Blue LED**: "BCM26"
+* **Button A**: "BCM21"
+* **Button B**: "BCM20"
+* **Button C**: "BCM16"
+* **Piezo buzzer**: "PWM1"
+* **Segment display**: "I2C1"
+* **LED strip**: "SPI0.0"
+* **Temperature Sensor**: "I2C1"
+* **Additional PWM**: "PWM0"
+
+This information will be useful later on, when we will ask you to manipulate some specific components.
+
+### Servo motor
+
+During this workshop, you will also use an additional hardware component: an **SG90 9g Micro Servo**.
+
+A Servo is a type of geared motor that can only rotate 180 degrees. It is controlled by sending electrical pulses that tell the servo what position it should move to.  
+A servo has three wires, the brown wire is GND, the red one is 5V, and the orange one is PWM.
+
+Now you know that, connect the servo to the following connectors of the RainbowHAT.
+
+[Spoiler: that way!][servo]
```

Comment:
Tip: You can actually plug the servo female pin directly to the rainbow hat PWM pins. Really interesting for our workshop, the cable we bought are finally useless 😄